tmva/sofie/inc/TMVA/ROperator_Cast.hxx

```diff
@@ -0,0 +1,73 @@
+#ifndef TMVA_SOFIE_ROPERATOR_Cast
+#define TMVA_SOFIE_ROPERATOR_Cast
+
+#include "TMVA/SOFIE_common.hxx"
+#include "TMVA/ROperator.hxx"
+#include "TMVA/RModel.hxx"
+
+#include <sstream>
+
+namespace TMVA{
+namespace Experimental{
+namespace SOFIE{
+
+
+template <typename T>
+class ROperator_Cast final : public ROperator
+{
+
+private:
+
+   std::string fNX;
+   std::string fNY;
+   std::vector<size_t> fShape;
+
+public:
+   ROperator_Cast(){}
+   ROperator_Cast(std::string nameX, std::string nameY):
+      fNX(UTILITY::Clean_name(nameX)), fNY(UTILITY::Clean_name(nameY)){}
+
+   std::vector<ETensorType> TypeInference(std::vector<ETensorType> input){
+      return input;
+   }
+
+   std::vector<std::vector<size_t>> ShapeInference(std::vector<std::vector<size_t>> input){
+      auto ret = input; //suggest copy to compiler
+      return ret;
+   }
+
+   void Initialize(RModel& model){
+       //input must be a graph input, or already initialized intermediate tensor
+      if (model.CheckIfTensorAlreadyExist(fNX) == false){
+        throw std::runtime_error("TMVA SOFIE Cast Op Input Tensor is not found in model");
+      }
+      fShape = model.GetTensorShape(fNX);
+      model.AddIntermediateTensor(fNY, model.GetTensorType(fNX), fShape);
+   }
+
+
+   std::string Generate(std::string OpName){
+      OpName = "op_" + OpName;
+      if (fShape.empty()) {
+         throw std::runtime_error("TMVA SOFIE Cast called to Generate without being initialized first");
+      }
+      std::stringstream out;
+      size_t length = ConvertShapeToLength(fShape);
+
+      out << "\n//------ CAST\n";
+      out << SP << "for (int id = 0; id < " << length << " ; id++){\n";
+
+      out << SP << SP << "tensor_" << fNY << "[id] = (float)(tensor_" << fNX << "[id]);\n";
```

Comment:
- We should use `static_cast<Type>` for casting 
- We should support casting to different types depending on the attribute of the operator.
The input type is determined is defined by the template parameter T